test/integration/test.mainpage.js

```diff
@@ -1,28 +1,120 @@
 /* eslint no-undef: 0 */
 /* eslint prefer-arrow-callback: 0 */
-casper.test.begin('Speech to Text', 7, function suite(test) {
+/**
+* @file
+*   Testing to verify the Speech to Text demo is working properly
+*/
+
+// Define the suite of tests and give it the following properties:
+// - Title, which shows up before any of the pass/fails.
+// - Number of tests, must be changed as you add tests.
+// - suite(), which contains all of your tests.
+//
+// @see http://casperjs.readthedocs.org/en/latest
+
+casper.test.begin('Speech to Text', 26, function suite(test) {
   const baseHost = 'http://localhost:3000';
 
   function testForButtons() {
-    casper.waitForSelector('div.buttons-container', function () {
-      test.assertExists('button.base--button.speak-button', 'displays speak button');
-      test.assertExists('button.base--button.download-button', 'displays download button');
-      test.assertExists('div.reset-container.dimmed', 'displays reset-container dimmed');
-      test.assertExists('div.reset-container.dimmed > a.reset-button', 'displays reset-container dimmed with child link');
+    casper.test.comment('Testing the 4 buttons');
+
+    test.assertExists('button.base--button:nth-child(7)', 'displays Record Audio');
+    test.assertExists('label.base--button', 'displays Upload Audio File');
+    test.assertExists('button.base--button:nth-child(9)', 'displays Play Sample 1');
+    test.assertExists('button.base--button:nth-child(10)', 'displays Play Sample 2');
+  }
+
+  function testVoiceModel() {
+    casper.test.comment('Testing the Voice Model box');
+
+    test.assertExists('select.base--select', 'displays Voice Model');
+    test.assertSelectorHasText('select.base--select option:nth-child(1)', 'French broadband', 'French model is found');
+    test.assertSelectorHasText('select.base--select option:nth-child(8)', 'Spanish broadband', 'Spanish model is found');
+  }
+
+  function testKeywords() {
+    casper.test.comment('Testing the Keywords to spot box');
+
+    test.assertExists('input#keywords.base--input', 'displays Keywords to spot');
+    test.assertSelectorHasText('input#keywords.base--input', 'changing the world', 'Changing the world is found');
```

Comment:
Not a blocker, but when using an ID (`#keywords`), you really don't need anything else to identify the element (`input` and `.base--input`). An ID must be unique across all elements on the page.